tenders/hvt/hvt_module_blk.c

```diff
@@ -106,32 +106,68 @@ static void hypercall_block_read(struct hvt *hvt, hvt_gpa_t gpa)
     rd->ret = SOLO5_R_OK;
 }
 
+#define BLOCK_PREFIX "--block:"
+#define BLOCK_SECTOR_SIZE_PREFIX "--block-sector-size:"
+
 static int handle_cmdarg(char *cmdarg, struct mft *mft)
 {
-    if (strncmp("--block:", cmdarg, 8) != 0)
+    enum {
+        opt_block,
+        opt_block_size,
+    } which;
+    if (strncmp(BLOCK_PREFIX, cmdarg, sizeof(BLOCK_PREFIX) - 1) == 0)
+        which = opt_block;
+    else if (strncmp(BLOCK_SECTOR_SIZE_PREFIX, cmdarg,
+                sizeof(BLOCK_SECTOR_SIZE_PREFIX) - 1) == 0)
+        which = opt_block_size;
+    else
         return -1;
 
     char name[MFT_NAME_SIZE];
     char path[PATH_MAX + 1];
-    int rc = sscanf(cmdarg,
-            "--block:%" XSTR(MFT_NAME_MAX) "[A-Za-z0-9]="
-            "%" XSTR(PATH_MAX) "s", name, path);
-    if (rc != 2)
-        return -1;
-    struct mft_entry *e = mft_get_by_name(mft, name, MFT_DEV_BLOCK_BASIC, NULL);
-    if (e == NULL) {
-        warnx("Resource not declared in manifest: '%s'", name);
-        return -1;
+    uint16_t block_size;
+    if (which == opt_block) {
+        block_size = 512;
+        int rc = sscanf(cmdarg,
+                BLOCK_PREFIX "%" XSTR(MFT_NAME_MAX) "[A-Za-z0-9]="
+                "%" XSTR(PATH_MAX) "s", name, path);
+        if (rc != 2)
+            return -1;
+
+        struct mft_entry *e = mft_get_by_name(mft, name, MFT_DEV_BLOCK_BASIC, NULL);
+        if (e == NULL) {
+            warnx("Resource not declared in manifest: '%s'", name);
+            return -1;
+        }
+        off_t capacity;
+        int fd = block_attach(path, block_size, &capacity);
```

Comment:
This is wrong. We don't know block_size yet. We should call `block_attach()` in `setup()`.

Reply:
Fixed in latest commit.